### PR TITLE
create evaluate module and remove ground truth support in inference

### LINKED ISSUE #249: Separate inference and benchmark (aka evaluate) modules
inference_segmentation.py currently offers to [optionally calculate metrics](https://github.com/NRCan/geo-deep-learning/blob/develop/inference_segmentation.py#L658) of a prediction on a new image, if the ground truth is inputted alongside. This feature is useful, but should belong to its own module. I suggest we remove support for ground truth at inference and plan a benchmark/evaluation module of its own.

### PR DESCRIPTION
**Pull request by remtav**
fix set crs to benchmark.gpkg:
- crs is set to original crs for each image
- once combined in final gpkg, a reprojection to 4326 is performed

closes #249

### CODE REVIEW

**Review by CharlesAuthier — DISMISSED** [review]

**Review by mpelchat04 — DISMISSED** [review]

File: evaluate_segmentation.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+            x_3, y_3 = x_2, (ymax - ((row * yres) + my))
+            x_4, y_4 = x_1, y_3
+            geom = Polygon([(x_1, y_1), (x_2, y_2), (x_3, y_3), (x_4, y_4)])
+            feature['geometry'].append(geom)
+            feature['length'].append(geom.length)
+            feature['pointx'].append(geom.centroid.x)
+            feature['pointy'].append(geom.centroid.y)
+            feature['area'].append(geom.area)
+            cnt += 1
+    gdf = gpd.GeoDataFrame(feature, crs=input_image.crs.to_epsg())
```
Comment (remtav) [inline evaluate_segmentation.py:67]: @victorlazio109 did you see my secret recipe to get the crs to work? :)
I think you had been struggling with this. I do 2 things:
1. set the epsg of each "inference" benchmark to the epsg of the original inference (i.e. of the imagery). The "to_epsg()" may have been something you were forgetting...
2. set all the benchmarking metrics to epsg 4326 if the inference span across Canada. (see comment below)

File: evaluate_segmentation.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+                                 fill=0,  # background value in rasterized vector.
+                                 attribute_values=attr_vals)
+        if debug:
+            logging.debug(f'\nUnique values in loaded label as raster: {np.unique(label)}\n'
+                          f'Shape of label as raster: {label.shape}')
+
+        gdf = metrics_per_tile(label_arr=label, pred_img=pred, input_image=raster_clipped, chunk_size=chunk_size,
+                               gpkg_name=local_gpkg.stem, num_classes=num_classes)
+
+        gdf_.append(gdf.to_crs(4326))
```
Comment (remtav) [inline evaluate_segmentation.py:162]: This is where the features are reprojected to lat/long. This step seems necessary before concatenating all geodataframes at line 172.

**Review by CharlesAuthier — DISMISSED** [review]

File: evaluate_segmentation.py (hunk 1 above)
Comment (victorlazio109) [reply evaluate_segmentation.py:67]: I think we had solved this issue previously, thanks to you!

File: evaluate_segmentation.py (hunk 2 above)
Comment (victorlazio109) [reply evaluate_segmentation.py:162]: Yeah, slightly different from what is implemented in inference script( see comment above). Confirm it works as intended

**Review by mpelchat04 — APPROVED** [review]

**Review by victorlazio109 — DISMISSED** [review]

**Review by victorlazio109 — APPROVED** [review]

**Review by mpelchat04 — APPROVED** [review]

**Pull request merged**